### PR TITLE
fix: exhibition of swap with zero balance

### PR DESCRIPTION
**Pull request by tuliomir**
### Acceptance Criteria
- Display `0.00` balance when a proposal has inputs and outputs for a token, but no change in the wallet balance.

### Visualization
Demonstration

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/utils/atomicSwap.js [hunk 1]
```diff
@@ -140,14 +140,14 @@ export function calculateExhibitionData(partialTx, cachedTokens, wallet) {
...
                 name: '',
-                status: PROPOSAL_DOWNLOAD_STATUS.LOADING
+                status: PROPOSAL_DOWNLOAD_STATUS.LOADING,
             };
         }
 
         return cachedTokens[tokenUid];
     }
 
-    const txProposal = new PartialTxProposal.fromPartialTx(partialTx.serialize(), wallet.getNetworkObject());
```
Comment (tuliomir) [inline src/utils/atomicSwap.js:150]: This is a bugfix: `fromPartialTx` is not a constructor and should not be called with `new`.

File: src/components/atomic-swap/ProposalBalanceTable.js [hunk 2]
```diff
@@ -15,12 +15,17 @@ import { t } from "ttag";
...
     const receivingBalances = balance.filter(b => b.receiving > 0);
+    // If the wallet participates in the proposal with a token with zero balance, it is displayed as "sending"
+    const sendingBalances = balance.filter(b => {
+        return b.sending > 0 || b.sending === -0;
+    });
 
     const renderRows = () => {
         const renderOne = (amount, symbol) => {
-            if (!amount || !symbol) return '';
+            if (!amount && !symbol) {
```
Comment (tuliomir) [inline src/components/atomic-swap/ProposalBalanceTable.js:26]: Javascript understands amount `0` as `false`, so we need to accept a _falsey_ amount that comes with a symbol.

File: src/components/atomic-swap/ProposalBalanceTable.js [hunk 3]
```diff
@@ -15,12 +15,17 @@ import { t } from "ttag";
  * @param {DisplayBalance[]} balance
  */
 export function ProposalBalanceTable ({ partialTx, wallet, balance }) {
-    const sendingBalances = balance.filter(b => b.sending > 0);
     const receivingBalances = balance.filter(b => b.receiving > 0);
+    // If the wallet participates in the proposal with a token with zero balance, it is displayed as "sending"
+    const sendingBalances = balance.filter(b => {
+        return b.sending > 0 || b.sending === -0;
```
Comment (r4mmer) [inline src/components/atomic-swap/ProposalBalanceTable.js:21]: -0? And why not `>=`?

Comment (tuliomir) [reply src/components/atomic-swap/ProposalBalanceTable.js]: This was a leftover from a previous implementation: simplified on [45499cf](https://github.com/HathorNetwork/hathor-wallet/pull/402/commits/45499cf8b842e173854c96db983188bcec317bd1).

**Review by r4mmer — APPROVED** [review]

**Review by alexruzenhack — APPROVED** [review]

**Pull request merged**